src/components/EventsList/EventsList.js

```diff
@@ -39,13 +41,29 @@ const EventsList = () => {
   }, []);
 
   useEffect(() => {
-    const filtered = events.filter((eventItem) => {
-      return (
-        eventItem.title.toLowerCase().includes(searchTerm.toLowerCase()) ||
-        eventItem.description.toLowerCase().includes(searchTerm.toLowerCase())
-      );
-    });
-    setSortedAndFilteredEvents(filtered);
+    let filtered = [...events];
+
+    if (searchTerm) {
+      filtered = filtered.filter((eventItem) => {
+        return (
+          eventItem.title
+            .toLowerCase()
+            .includes(searchTerm.toLowerCase().trim()) ||
+          eventItem.description
+            .toLowerCase()
+            .includes(searchTerm.toLowerCase().trim())
+        );
+      });
+    }
+
+    let sortedAndFiltered = [...filtered];
+
+    if (sortValue !== "none") {
+      sortedAndFiltered = getSortedEvents(sortedAndFiltered, sortValue);
+      setSortedAndFilteredEvents(sortedAndFiltered);
+    } else {
+      setSortedAndFilteredEvents(filtered);
+    }
```

Comment:
you don't need an "else" here, since sortedAndFiltered had value filtered before the if. You can just setSortedAndFilteredEvents(sortedAndFiltered) outside the if